cpp/inc/bond/protocol/encoding.h

```diff
@@ -118,12 +118,15 @@ ReadVariableUnsigned(Buffer& input, T& value)
 }
 
 
-// If protocol's Write(const bond::blob&) method doesn't write raw blob to
-// output, this function needs to be overloaded appropriately.
-template <typename Writer>
-inline void WriteRawBlob(Writer& writer, const blob& data)
-{
-    writer.Write(data);
+// Returns an object that represents a buffer range. The input arguments are
+// determined by what the GetCurrentBuffer returns for the given input buffer
+// implementation (i.e. not necessarily a blob). The GetBufferRange may return
+// a type different from blob as long as it is understood by the corresponding
+// output buffer associated with the writer (i.e. there is an overloaded Write
+// function that accepts it).
+inline blob GetBufferRange(const blob& begin, const blob& end)
```

Comment:
Should there be a generic definition/declaration for `GetBufferRange`, `CreateInputBuffer`, and `CreateOutputBuffer` to a) provide a good failure message when they are unimplemented and b) show the concept that each needs to implement?